### PR TITLE
Update Charms to allow repairs on standard anvil.

### PR DESCRIPTION
**Pull request by trunksbomb**
Update Charms to allow repairs on standard anvil with their crafting item (Blaze Rod for Fire Charm, Emerald for Speed Charm, etc).

Addresses Issue #956

### CODE REVIEW

**Comment by trunksbomb** [comment]
I'm enjoying your mod as part of All the Mods 3 modpack! Noticed that you could not repair charms on the anvil and I saw that your code was very clean and concise so I added that feature.

Sorry about the multiple commits, I missed the new interface class in the first commit and had to add it.

Tested on Minecraft 1.12.2 and Forge 1.4.23.5.2768

**Comment by Lothrazar** [comment]
Hey this looks good ill check it out

File: src/main/java/com/lothrazar/cyclicmagic/item/equipbauble/ItemCharmAir.java [hunk 1]
```diff
@@ -96,7 +97,13 @@ public void onTick(ItemStack stack, EntityPlayer player) {
   }
 
   @Override
-  public IRecipe addRecipe() {
-    return super.addRecipe(new ItemStack(Blocks.BONE_BLOCK));
+  public boolean getIsRepairable(ItemStack par1ItemStack, ItemStack par2ItemStack)
+  {
+    return par2ItemStack.getItem() == craftItem.getItem();
```
Comment (Lothrazar) [inline src/main/java/com/lothrazar/cyclicmagic/item/equipbauble/ItemCharmAir.java]: This doesnt consider metadata or nbt so for example it woill match coal and charcoal.  . which is ok for the simple items we're using, but ill probably just change it after merge

File: src/main/java/com/lothrazar/cyclicmagic/item/equipbauble/ItemCharmAntidote.java [hunk 2]
```diff
@@ -37,9 +37,10 @@
 import net.minecraft.item.crafting.IRecipe;
 import net.minecraftforge.common.config.Configuration;
 
-public class ItemCharmAntidote extends BaseCharm implements IHasRecipe, IContent {
+public class ItemCharmAntidote extends BaseCharm implements IHasRecipeAndRepair, IContent {
 
   private static final int durability = 32;
+  private static final ItemStack craftItem = new ItemStack(Items.FERMENTED_SPIDER_EYE);
```
Comment (Lothrazar) [inline src/main/java/com/lothrazar/cyclicmagic/item/equipbauble/ItemCharmAntidote.java]: Good call refactoring this to a static

**Comment by Lothrazar** [comment]
refactored to make interface extend the other recipe one but is very good pr thanks for this, they are always welcome

**Pull request merged**